spotify.go

```diff
@@ -124,6 +175,74 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+	retryDuration time.Duration
+}
+
+func (c *Client) ExecuteOpt(req *http.Request, needsStatus int, result interface{}) (err error) {
+	for {
+		resp, err := c.http.Do(req)
+		if err != nil {
+			return err
+		}
+
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK || needsStatus != 0 && resp.StatusCode != needsStatus {
+			errorMessage := decodeError(c, resp)
+
+			if errorMessage.Error() == rateLimitExceededErrorMessage && autoRetry {
+				time.Sleep(c.retryDuration)
+				continue
+			}
+
+			return errorMessage
+		}
+
+		if result != nil {
+			if err := json.NewDecoder(resp.Body).Decode(result); err != nil {
+				return err
+			}
+		}
+
+		break
+	}
+
+	return nil
+}
+
+func (c *Client) Execute(req *http.Request) (err error) {
```

Comment:
Done, though named return-values are used in the `NewReleases` methods.

I'll update these calls to not be exported in the next few minutes.